herbert_core/classes/MPIFramework/@MessagesCppMPI/MessagesCppMPI.m

```diff
@@ -221,6 +221,16 @@ function delete(obj)
             %       failure.
             % >> on success, message contains an object of class aMessage,
             %        with the received message contents.
+            %NOTE:
+            % When synchronous receive is selected, receive waits for
+            % the specific message with specific data tag to be send.
+            % If interrupt message appears after the framework starts
+            % waiting for a synchronous message, the framework hangs up.
+            % Receive_all should be used to avoid such hang ups.
```

Comment:
Is there any use case for not using `receive_all` if as you describe here the framework will hang up if any async message is received while waiting for a blocking one?

If there isn't should these non-`receive-all` methods simply be removed or made private? Having an API that we *know* will cause the framework to fail feels like a high risk strategy.

Reply:
as discussed, only interrupt will hang up the framework as no data message will be send after it. An asynchronous message will be successfully received during next appropriate receive operation, so this behavior constrains only error processing. But this is weak constrain, slightly restricting transmission of the debug information, so I think this behavior should be kept as important optimization tool.